executor/src/witgen/mod.rs

```diff
@@ -454,6 +471,7 @@ impl<'a, T: FieldElement> FixedData<'a, T> {
             challenges,
             global_range_constraints,
             intermediate_definitions,
+            stage,
```

Comment:
So in `FixedData`, the only members which depend on the stage are `stage` and `witness_cols`?

Reply:
It would almost feel clearer to have a different struct for these, so that FixedData can always be the same? Because for example now it seems like `column_by_name` has keys which are not found in `witness_cols`

Reply:
I'm not too opposed to creating a `FixedData` per stage, but we should clarify what we do with lower-stage witness columns.

Reply:
`witness_cols` contains all columns, regardless of the stage! The only new thing is that it stores the stage.

Reply:
That wouldn't actually be trivial to change, because we do:
```rs
let witness_cols = WitnessColumnMap::from(
    analyzed.committed_polys_in_source_order().flat_map(
        |(symbol, value)| { symbol.array_elements()...
```
But a `WitnessColumnMap` can't have any gaps.

Reply:
Oh my bad, I misread the `flat_map` in the constructor as a `filter_map`.

Reply:
Feel free to close this